repos/system_upgrade/el7toel8/actors/migratentp/libraries/ntp2chrony.py

```diff
@@ -0,0 +1,664 @@
+        if not os.path.isfile(path):
+            if self.verbose > 0:
+                print("Missing " + path)
+            return
+
+        with open(path, encoding="latin-1") as f:
+            for line in f:
+                line = line[:line.find('#')]
+
+                words = line.split()
+
+                if not words:
+                    continue
+
+                self.step_tickers.extend(words)
+
+    def parse_ntp_conf(self, path=None):
+        if path is None:
+            path = self.root_dir + self.ntp_conf_path
+
+        with open(path, encoding="latin-1") as f:
+            if self.verbose > 0:
+                print("Reading " + path)
+
+            for line in f:
+                line = line[:line.find('#')]
+
+                words = line.split()
+
+                if not words:
+                    continue
+
+                if not self.parse_directive(words):
+                    self.ignored_lines.append(line)
+
+    def parse_directive(self, words):
+        name = words.pop(0)
+        if name.startswith("logconfig"):
+            name = "logconfig"
+
+        if words:
+            if name in ["server", "peer", "pool"]:
+                return self.parse_source(name, words)
+            elif name == "fudge":
+                return self.parse_fudge(words)
+            elif name == "restrict":
+                return self.parse_restrict(words)
+            elif name == "tos":
+                return self.parse_tos(words)
+            elif name == "includefile":
+                return self.parse_includefile(words)
+            elif name == "keys":
+                return self.parse_keys(words)
+            elif name == "trustedkey":
+                return self.parse_trustedkey(words)
+            elif name == "driftfile":
+                self.driftfile = words[0]
+            elif name == "statistics":
+                self.statistics = words
+            elif name == "leapfile":
+                self.leapfile = words[0]
+            else:
+                self.ignored_directives.add(name)
+                return False
+        else:
+            self.ignored_directives.add(name)
+            return False
+
+        return True
+
+    def parse_source(self, type, words):
+        ipv4_only = False
+        ipv6_only = False
+        source = {
+                "type": type,
+                "options": []
+        }
+
+        if words[0] == "-4":
+            ipv4_only = True
+            words.pop(0)
+        elif words[0] == "-6":
+            ipv6_only = True
+            words.pop(0)
+
+        if not words:
+            return False
+
+        source["address"] = words.pop(0)
+
+        # Check if -4/-6 corresponds to the address and ignore hostnames
+        if ipv4_only or ipv6_only:
+            try:
+                version = ipaddress.ip_address(source["address"]).version
+                if (ipv4_only and version != 4) or (ipv6_only and version != 6):
+                    return False
```

Comment:
same thing here, please see comment for changing return False to return a_dict_of_processed_values

Reply:
Is it a bug in time source parsing with include file https://github.com/mlichvar/ntp2chrony/blob/master/tests/data/chronyconfs/3_chrony.conf#L4? Several lines appear twice in generated config.